### PR TITLE
Add ability to use custom CNAME for Customer Web Server public

### PR DESCRIPTION
**Pull request by spirosoik**
#### Summary
Ability to pass custom CNAME instead of public nginx hostname for Customer Web Server

#### Ticket Link
Ticket: https://mattermost.atlassian.net/browse/MM-38605

### CODE REVIEW

**Review by stafot — APPROVED** [review]

**Review by stylianosrigas — APPROVED** [review]
LGTM

File: terraform/aws/modules/route53-registration/variables.tf [hunk 1]
```diff
@@ -1,19 +1,54 @@
...
+}
+
+variable "enable_portal_private_r53_record" {
+  type        = bool
+  description = "Enables to create a private CNAME route53 record for Private Customer Web Server"
+}
+
+variable "enable_portal_internal_r53_record" {
+  type        = bool
+  description = "Enables to create a internal CNAME route53 record for Internal Customer Web Serve APIr"
```
Comment (angeloskyratzakos) [inline terraform/aws/modules/route53-registration/variables.tf:33]: tiny typo at the end
```suggestion
  description = "Enables to create a internal CNAME route53 record for Internal Customer Web Serve API"
```

File: terraform/aws/modules/route53-registration/variables.tf [hunk 2]
```diff
@@ -1,19 +1,54 @@
...
+  description = "The Cloudflare zone ID provided"
+}
+
+variable "enable_portal_public_r53_record" {
+  type        = bool
+  description = "Enables to create a public route53 record for public Customer Web Server"
+}
+
+variable "enabled_cloudflare_customer_web_server" {
+  type        = string
```
Comment (angeloskyratzakos) [inline terraform/aws/modules/route53-registration/variables.tf:22]: is this a string or bool?

Comment (angeloskyratzakos) [reply terraform/aws/modules/route53-registration/variables.tf:22]: if this changes to bool the if statement above also needs to change

Comment (spirosoik) [reply terraform/aws/modules/route53-registration/variables.tf:22]: correct this is a bool. nice catch

**Review by angeloskyratzakos — APPROVED** [review]
LGTM

File: terraform/aws/modules/route53-registration/route53.tf [hunk 3]
```diff
@@ -79,14 +79,29 @@ resource "aws_route53_record" "awat" {
...
+
 resource "aws_route53_record" "customer_web_server" {
   count = var.enable_portal_public_r53_record ? 1 : 0
 
   zone_id = var.public_hosted_zoneid
   name    = "portal"
   type    = "CNAME"
   ttl     = "60"
-  records = [data.kubernetes_service.nginx-public.status.0.load_balancer.0.ingress.0.hostname]
+  records = [var.enabled_cloudflare_customer_web_server == "" ? data.kubernetes_service.nginx-public.status.0.load_balancer.0.ingress.0.hostname : cloudflare_record.customer_web_server_cname.hostname]
```
Comment (angeloskyratzakos) [inline terraform/aws/modules/route53-registration/route53.tf:104]: if changed to bool
```suggestion
  records = [var.enabled_cloudflare_customer_web_server ? data.kubernetes_service.nginx-public.status.0.load_balancer.0.ingress.0.hostname : cloudflare_record.customer_web_server_cname.hostname]
```

**Pull request merged**